test/rewards/AutoCompoundingRewards.ts

```diff
@@ -686,6 +737,85 @@ describe('AutoCompoundingRewards', () => {
+                        { tokenSymbol: TokenSymbol.TKN3, initialUserStake: toWei(60_000), totalRewards: toWei(16_000) },
+                        { tokenSymbol: TokenSymbol.TKN4, initialUserStake: toWei(70_000), totalRewards: toWei(17_000) },
+                        { tokenSymbol: TokenSymbol.TKN5, initialUserStake: toWei(80_000), totalRewards: toWei(18_000) }
+                    ];
+
+                    const tokens: TokenWithAddress[] = new Array<TokenWithAddress>(setups.length);
+                    const poolTokens: TokenWithAddress[] = new Array<TokenWithAddress>(setups.length);
+
+                    beforeEach(async () => {
+                        for (const [index, setup] of setups.entries()) {
+                            ({ token: tokens[index], poolToken: poolTokens[index] } = await prepareSimplePool(
+                                new TokenData(setup.tokenSymbol),
+                                setup.initialUserStake,
+                                setup.totalRewards
+                            ));
+
+                            await createProgram(
+                                distributionType,
+                                autoCompoundingRewards,
+                                tokens[index].address,
+                                setup.totalRewards,
+                                START_TIME
+                            );
+                        }
+
+                        await autoCompoundingRewards.setAutoProcessRewardsCount(AUTO_PROCESS_REWARDS_COUNT);
+                    });
+
+                    if (distributionType === RewardsDistributionType.Flat) {
+                        it('should distribute all tokens', async () => {
+                            await autoCompoundingRewards.setTime(programEndTime[distributionType]);
+                            for (let i = 0; i < Math.ceil(setups.length / AUTO_PROCESS_REWARDS_COUNT); i++) {
+                                const res = await autoCompoundingRewards.autoProcessRewards();
+                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                            }
+                            const res = await autoCompoundingRewards.autoProcessRewards();
+                            await expect(res).not.to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                        });
+
+                        it('should distribute some tokens', async () => {
+                            for (let i = 0; i < 5; i++) {
+                                await autoCompoundingRewards.setTime(
+                                    Math.floor(START_TIME + (programDuration[distributionType] * 2 ** i) / (2 ** i + 1))
+                                );
+                                const res = await autoCompoundingRewards.autoProcessRewards();
+                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
```

Comment:
FYI: my baseline for these tests was the 'process rewards' test group, where event parameters aren't tested.

Reply:
Yeah, I remember that you removed the explicit param checks since it was too complicated to do it on time back then and also since it wasn't critical to the test, but in this case - it is. There may be only a single emitted `RewardsDistributed` event with random parameters, and your test assumes, for some reason, that it was emitted 10 times for the correct pool. 

If you want to rely on the event for this test (there are other options, but this one isn't necessarily a bad one either), you can still enumerate and check all the events manually - but only check the first `Token indexed pool` parameter (manually).